lib/tests/misc.nix

```diff
@@ -312,6 +312,49 @@ runTests {
     expected = true;
   };
 
+  testToInt = testAllTrue [
+    # Naive
+    (123 == toInt "123")
+    (0 == toInt "0")
+    # Whitespace Padding
+    (123 == toInt " 123")
+    (123 == toInt "123 ")
+    (123 == toInt " 123 ")
+    (123 == toInt "   123   ")
+    (0 == toInt " 0")
+    (0 == toInt "0 ")
+    (0 == toInt " 0 ")
+    # Zero Padding
+    (123 == toInt "0123")
+    (123 == toInt "0000123")
+    (0 == toInt "000000")
+    # Whitespace and Zero Padding
+    (123 == toInt " 0123")
+    (123 == toInt "0123 ")
+    (123 == toInt " 0123 ")
+    (123 == toInt " 0000123")
+    (123 == toInt "0000123 ")
+    (123 == toInt " 0000123 ")
+    (0 == toInt " 000000")
+    (0 == toInt "000000 ")
+    (0 == toInt " 000000 ")
+  ];
+
+  #testToIntFailsOnEmptyString = {
+  #  expr = builtins.tryEval (toInt "");
+  #  expected = { success = false; value = false; };
+  #};
```

Comment:
This is the commented out test. If uncommented, it causes errors to escape the tryEval.